packages/aws-rfdk/lib/deadline/scripts/python/test_configure_identity_registration_settings.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env python3
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Tests for configure_identity_registration_settings.py
+"""
+
+import sys
+import unittest
+from unittest.mock import MagicMock, Mock
+
+sys.modules['boto3'] = Mock()
+import configure_identity_registration_settings as subject
+
+USERNAME = "admin"
+PASSWORD = "Test567#"
+DEADLINE_COMMAND_CLIENT = Mock()
+
+REG_SETTING_RFDK_1 = {
+    "ConnectionIpFilterType":0,
+    "ConnectionIpFilterValue":"111.111.*.*", # 111.111.0.0/16
+    "SourceIpFilterType":0,
+    "SourceIpFilterValue":"123.123.123.*", # 123.123.123.0/24
+    "SettingsId":"settings1",
+    "SettingsName":"RfdkSubnet|subnet-aaaaaaa|subnet-1111111",
+    "IsEnabled":True,
+    "DefaultStatus":2,
+    "DefaultRole":"Client"
+}
```

Comment:
nit: could these be made into their named tuple literal equiavlents?

```suggestion
REG_SETTING_RFDK_1 = subject.LoadBalancerIdentityRegistrationSetting(
    connection_ip_filter_type=0,
    connection_ip_filter_value="111.111.*.*", # 111.111.0.0/16
    source_ip_filter_type=0,
    source_ip_filter_value="123.123.123.*", # 123.123.123.0/24
    settings_id="settings1",
    settings_name="RfdkSubnet|subnet-aaaaaaa|subnet-1111111",
    is_dnabled=True,
    default_status=2,
    default_role="Client"
)
```

this de-couples the tests from the `LoadBalancerIdentityRegistrationSetting.from_json(...)`. Same feedback applies to other places in this file

Reply:
Updated

Reply:
I'd overlooked the fact that `get_rfdk_registration_settings` gets JSON-encoded settings from `deadlinecommand` and passes that to `LoadBalancerIdentityRegistrationSetting.from_json`. This makes testing tricky since we either need to mock `deadlinecommand` to return valid JSON, or mock `LoadBalancerIdentityRegistrationSetting.from_json` to have stateful returns.

Either is not ideal, but I think it suffices for now and if we ever need to, we can refactor that later. Resolving.